### PR TITLE
sql: document SPLIT AT statement

### LINKED ISSUE #616: sql: SPLIT AT statement
PR: https://github.com/cockroachdb/cockroach/pull/8938

### PR DESCRIPTION
**Pull request by nvanbenschoten**
Fixes #616.

This change documents the `ALTER [TABLE|INDEX] .. SPLIT AT ..`
statement. It describes how to use the statement and then provides
justification for why an operator might want to use it.

Depends on https://github.com/cockroachdb/cockroach/pull/23721.

### CODE REVIEW

**Comment by benesch** [comment]
If you rebase this you'll pick up the new speedy CI link checker. Looks like the old link checker hit a spurious timeout.

File: v2.0/split-at.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+key-space known as ranges. By default, CockroachDB attempts to keep ranges below
+a size of 64MiB. To do this, the system will automatically [split](architecture/distribution-layer.html#range-splits)
+a range if it grows larger than this limit. For most use cases, this automatic
+range splitting is sufficient and a user should never need to worry about
+when or where the system decides to split ranges.
+
+However, there are reasons why operators may want to perform manual splits on
+the ranges that store tables or indexes:
+
+- When a range only consists of a single range, all writes and reads to the
```
Comment (jseldess) [inline v2.0/split-at.md]: `When a range` > `When a table`

File: v2.0/split-at.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  be distributed across multiple nodes and this will not be a concern.
+
+- When a table is created, it will only consist of a single range. If an
+  operator knows that a new table will immediately receive significant write
+  traffic, they may want to preemptively split the table based on the expected
+  distribution of writes before applying the load. This can help avoid reduced
+  workload performance that results when automatic splits are unable to keep up
+  with write traffic.
+
+- TODO blah blah something about follow the workload.. Does this undermine our
```
Comment (jseldess) [inline v2.0/split-at.md]: @nvanbenschoten, what did you want to write about follow the workload here?

File: v2.0/split-at.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  messaging about PARTITIONING?
+
+## Example
```
Comment (jseldess) [inline v2.0/split-at.md]: nit: `Examples`

File: v2.0/split-at.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
+---
+title: SPLIT AT
+summary: The SPLIT AT forces a key-value layer range split at the specified row in a table or index.
```
Comment (jseldess) [inline v2.0/split-at.md]: nit: `The SPLIT AT statement...`

Comment (jseldess) [reply v2.0/split-at.md]: Done.

File: v2.0/split-at.md (hunk 1 above)
Comment (jseldess) [reply v2.0/split-at.md]: Done.

File: v2.0/split-at.md (hunk 3 above)
Comment (jseldess) [reply v2.0/split-at.md]: Done.

**Pull request merged**